jina/executors/indexers/query/vector.py

```diff
@@ -0,0 +1,96 @@
+from typing import Generator, Iterable, Optional, Dict
+
+import numpy as np
+
+from jina.executors.dump import import_vectors
+from jina.executors.indexers.query import BaseQueryIndexer
+from jina.executors.indexers.vector import NumpyIndexer
+
+
+class QueryNumpyIndexer(NumpyIndexer, BaseQueryIndexer):
+    """An exhaustive vector indexers implemented with numpy and scipy.
+
+    .. note::
+        Metrics other than `cosine` and `euclidean` requires ``scipy`` installed.
+
+    :param metric: The distance metric to use. `braycurtis`, `canberra`, `chebyshev`, `cityblock`, `correlation`,
+                    `cosine`, `dice`, `euclidean`, `hamming`, `jaccard`, `jensenshannon`, `kulsinski`,
+                    `mahalanobis`,
+                    `matching`, `minkowski`, `rogerstanimoto`, `russellrao`, `seuclidean`, `sokalmichener`,
+                    `sokalsneath`, `sqeuclidean`, `wminkowski`, `yule`.
+    :param backend: `numpy` or `scipy`, `numpy` only supports `euclidean` and `cosine` distance
+    :param compress_level: compression level to use
+    """
+
+    def _post_init_wrapper(
+        self,
+        _metas: Optional[Dict] = None,
+        _requests: Optional[Dict] = None,
+        fill_in_metas: bool = True,
+    ) -> None:
+        super()._post_init_wrapper(_metas, _requests, fill_in_metas)
+        dump_path = _metas.get('dump_path')
+        if dump_path:
+            self.load_dump(dump_path)
+
+    def load_dump(self, dump_path):
+        """Load the dump at the path
+
+        :param dump_path: the path of the dump"""
+        ids, vecs = import_vectors(dump_path, str(self.pea_id))
+        self.add(ids, vecs)
+        self.write_handler.flush()
+        self.write_handler.close()
+        self.handler_mutex = False
+        self.is_handler_loaded = False
+        test_vecs = np.array([np.random.random(self.num_dim)], dtype=self.dtype)
+        assert self.query(test_vecs, 1) is not None
+        self.add = lambda *args, **kwargs: self.logger.warning(
+            f'Index {self.index_abspath} is write-once'
+        )
+
+    def add(self, keys: Generator, vectors: Generator, *args, **kwargs) -> None:
```

Comment:
rename to `_add` or even use a slightly different naming, if you have a better name at hand.

Reply:
Yes, I removed the `add` from `QueryIndexers` and am only using the private `_add`